src/mrpro/operators/Functional.py

```diff
@@ -0,0 +1,64 @@
+"""Base Class Functional."""
+
+# Copyright 2024 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from abc import abstractmethod
+from collections.abc import Sequence
+
+import torch
+
+from mrpro.operators.Operator import Operator
+
+
+class Functional(Operator[torch.Tensor, tuple[torch.Tensor]]):
+    """Functional Base Class."""
+
+    def __init__(
+        self, weight: torch.Tensor | float = 1.0, target: torch.Tensor | None = None, dim: Sequence[int] | None = None, divide_by_n : bool = False
```

Comment:
do we really need weight, target and divide_by_n in the base class? This sounds like something that we would typically want for a norm but seems unintuitive for me for a general functional? At least I would struggle to write a formal description of what these tensors are supposed to be in general without having to refer to norms/data discrepancy/similarity measures as a more specific example of functionals.

Reply:
These are settings that should exist for all functionals.

You should be able to depend upon that fact.
Also, setting them is the same for all functionals l.

So I would say they are part of the base class.

You will never see the doc string of the base class anyway.

Reply:
What is the issue with 
F:Kn->R
Functional(•):= F(weight*(•-target)) / n ?

Reply:
> What is the issue with F:Kn->R Functional(•):= F(weight*(•-target)) / n ?

Just that it sounds a bit arbitrary to provide the base class with those specific choices of hyper-parameters that define the functional.

Would it be a problem to have a base class `functional `that inherits from `operator `and to have a base class `norm` that inherits from `functional ` and for which one can provide `weight`, `target `and `divide_by_n`?